js/views/file_input_view.js

```diff
@@ -1,236 +1,314 @@
-/* eslint-disable */
-
 /* global textsecure: false */
-
+/* global Whisper: false */
+/* global i18n: false */
+/* global loadImage: false */
+/* global Backbone: false */
+/* global _: false */
+/* global Signal: false */
+
+// eslint-disable-next-line func-names
 (function () {
-    'use strict';
-    window.Whisper = window.Whisper || {};
-
-    const { MIME } = window.Signal.Types;
-
-    Whisper.FileSizeToast = Whisper.ToastView.extend({
-        templateName: 'file-size-modal',
-        render_attributes: function() {
-            return {
-                'file-size-warning': i18n('fileSizeWarning'),
-                limit: this.model.limit,
-                units: this.model.units
-            };
-        }
+  'use strict';
+
+  window.Whisper = window.Whisper || {};
+
+  const { MIME } = window.Signal.Types;
+
+  Whisper.FileSizeToast = Whisper.ToastView.extend({
+    templateName: 'file-size-modal',
+    render_attributes() {
+      return {
+        'file-size-warning': i18n('fileSizeWarning'),
+        limit: this.model.limit,
+        units: this.model.units,
+      };
+    },
+  });
+  Whisper.UnsupportedFileTypeToast = Whisper.ToastView.extend({
+    template: i18n('unsupportedFileType'),
+  });
+
+  function makeThumbnail(size, objectUrl) {
+    return new Promise(((resolve, reject) => {
+      const img = document.createElement('img');
+      img.onerror = reject;
+      img.onload = () => {
+        // using components/blueimp-load-image
+
+        // first, make the correct size
+        let canvas = loadImage.scale(img, {
+          canvas: true,
+          cover: true,
+          maxWidth: size,
+          maxHeight: size,
+          minWidth: size,
+          minHeight: size,
+        });
+
+        // then crop
+        canvas = loadImage.scale(canvas, {
+          canvas: true,
+          crop: true,
+          maxWidth: size,
+          maxHeight: size,
+          minWidth: size,
+          minHeight: size,
+        });
+
+        const blob = window.dataURLToBlobSync(canvas.toDataURL('image/png'));
+
+        resolve(blob);
+      };
+      img.src = objectUrl;
+    }));
+  }
+
+  function makeVideoScreenshot(objectUrl) {
```

Comment:
Nice work on figuring this out! Was it ok to take the screenshot from the default frame the browser provided? Moxie said he’d fast-forward a hundred milliseconds or so to skip over any fade from black intros, etc.

Reply:
Today it matches what you see in a message when the video player is hosted for an attachment (before you press play). I feel like that consistency is probably worth keeping it the way it is. If you feel strongly, we could absolutely start the video playing and capture a screenshot later.